### DISCUSSION TITLE
Getting EXECUTION_LIMIT_REACHED while doing onchain transactions

### DISCUSSION THREAD

**Discussion by shivam2320** [aptos-labs/aptos-developer-discussions#170] [General]
I'm doing some onchain transactions on testnet but getting EXECUTION_LIMIT_REACHED in wallet

Few minutes back I was able to call the function but now I'm getting this error

I see this error is because of max gas limit per block and I need to call pyth multiple times for getting onchain prices of different assets, is this causing issue if yes how can I solve the same

**Comment by gregnazario** [comment]
> I'm doing some onchain transactions on testnet but getting EXECUTION_LIMIT_REACHED in wallet
> 
> Few minutes back I was able to call the function but now I'm getting this error
> 
> I see this error is because of max gas limit per block and I need to call pyth multiple times for getting onchain prices of different assets, is this causing issue if yes how can I solve the same

What's your code look like?  Does it do an increasing amount of work over time, that leads to increased gas usage?

**Reply by shivam2320** [reply]
It can increase amount of work but as of now there is same amount of work for all calls

**Reply by shivam2320** [reply]
But is mas gas limit per block main reason for getting EXECUTION_LIMIT_REACHED error

**Reply by mshakeg** [reply]
Hi @gregnazario I'm getting a similar `EXECUTION_LIMIT_REACHED` issue to [this tx](https://aptoscan.com/transaction/558757159). I'm not specifying the `maxGasAmount`, so presumably the sdk autopopulates that field from the simulation result/estimation with `200_000`. The tx is quite heavy, so it's possible that `200_000` gas units is not sufficient, is the max gas units for a tx really `930_000_000`? that seems really high, if I specify that value and only `1_000_000` gas units are actually used am I only billed for the `1_000_000` used or all `930_000_000`?
